core/src/main/codegen/templates/Parser.jj

```diff
@@ -1723,7 +1723,7 @@ SqlNode SqlMerge() :
     final Span s;
 }
 {
-    <MERGE> { s = span(); } <INTO> table = TableRefWithHintsOpt()
+    (<MERGE_INTO> { s = span(); } | <MERGE> { s = span(); } <INTO>) table = TableRefWithHintsOpt()
```

Comment:
To clarify, there should be we add extra parentheses to ensure INTO can't match with MERGE_INTO or is that clearly not an issue?

Reply:
I believe the syntax is fine as it is, but I'll add an extra set of parentheses for safety.